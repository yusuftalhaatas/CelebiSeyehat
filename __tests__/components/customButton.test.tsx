import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CustomButton from '../../src/components/customButton';

describe('components/CustomButton', () => {
  test('renders correctly and responds to press events', () => {
    const onPressMock = jest.fn();
    const {getByText, toJSON} = render(
      <CustomButton title="Test Button" onPress={onPressMock} />,
    );

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // onPress event test
    fireEvent.press(getByText('Test Button'));
    expect(onPressMock).toHaveBeenCalled();
  });

  test('renders correctly when disabled', () => {
    const onPressMock = jest.fn();
    const {getByText, toJSON} = render(
      <CustomButton title="Test Button" onPress={onPressMock} disabled />,
    );

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // onPress event should not trigger
    fireEvent.press(getByText('Test Button'));
    expect(onPressMock).not.toHaveBeenCalled();
  });
});
