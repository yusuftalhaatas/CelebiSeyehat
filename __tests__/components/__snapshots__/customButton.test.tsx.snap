// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CustomButton renders correctly and responds to press events 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
        
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#00ccff",
        "borderRadius": 17,
        "flexDirection": "column",
        "height": 50,
        "justifyContent": "center",
        "marginHorizontal": 50,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`components/CustomButton renders correctly when disabled 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#00ccff",
        "borderRadius": 17,
        "flexDirection": "column",
        "height": 50,
        "justifyContent": "center",
        "marginHorizontal": 50,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
